Blocks/Persistence/file.h

```diff
@@ -203,6 +218,9 @@ class FilePersister {
         // The `next.us` stores the closest possible next entry timestamp,
         // so the last processed entry timestamp is always 1us less.
         end.store({next.index, next.us - std::chrono::microseconds(1), head});
+        if (!current_offset) {
```

Comment:
Need a comment here on the value of this condition. I'm *guessing* it's "if no entries have been appended yet", but it's only my guess, which is not good enough for a random reader.